### PR TITLE
Adding instances for non-JSON content types to `servant-foreign`

### LINKED ISSUE #509: (servant-foreign / servant-js) missing instances for PlainText content-type
ghc gives
```
No instance for servant-foreign-0.6.1:Servant.Foreign.Internal.NotFound
  arising from a use of ‘jsForAPI’
In the second argument of ‘($)’, namely ‘jsForAPI api vanillaJS’
In the first argument of ‘(:<|>)’, namely
  ‘(return $ jsForAPI api vanillaJS)’
In the second argument of ‘(:<|>)’, namely
  ‘(return $ jsForAPI api vanillaJS)
   :<|> (return $ decodeUtf8 . toStrict . encode . toSwagger $ api)’
```

when I try to make js output for ("api" :> Get '[PlainText] Text)

### PR DESCRIPTION
**Pull request by tmbull**
I am trying to fix #509 (and probably #551). I have only done the work for `Verb` thus far, but I wanted some feedback before I proceed with `ReqBody`. 

In this first approach, I have just modified `Req` to include a list of `MediaType` that comes from the Accept class. From the perspective of someone wanting to use `servant-foreign`, I'm not sure if this is the correct approach. It might be better to create a type similar to `HeaderArg` or create a typeclass separate from `Accept` specifically for this purpose.  Thoughts?

### CODE REVIEW

**Comment by alpmestan** [comment]
This seems quite reasonable to me, even though I have not followed servant-foreign closely for the past year, so maybe someone from @haskell-servant/maintainers more familiar with it than me can take a look?

**Comment by tmbull** [comment]
Per @alpmestan's feedback, I've gone ahead and made the same changes to the `Req` type for `ReqBody`. The advantage I see of this approach is that it works for any type that implements `Accept`. Let me know what you guys think.

**Comment by alpmestan** [comment]
I'm pretty happy with what I see in the patch. At this point maybe the other thing we should discuss and consider is: have you thought already about "the other side", i.e how easy or hard it is to use this for doing codegen in other languages in a way that's handy for people to use?

**Comment by tmbull** [comment]
[Here ](https://github.com/tmbull/servant-elm/commit/ce848d0a81c251de781715f5913b0ca35ad4ce64#diff-7e13462cc66d19b6a0d2a0ddcd7e8864R384) is a link to my corresponding commit to `servant-elm` and here's a snippet: 
```
...
  expect =
      if
        any
        (\m -> m == F.contentType (Proxy :: Proxy F.JSON))
        (request ^. F.reqReturnContentTypes)
      then
        case request ^. F.reqReturnType of
...
```

Basically, I can just check if the expected content type is in the results. Calling `contentType` on the `Proxy` type allowed me to avoid using string literals. I do not have experience with any other codegen libraries that use `servant-foreign`, but for my purpose this seems reasonable to me. What do you guys think?

**Comment by alpmestan** [comment]
Looks reasonable to me. Let's give a little bit more time (a few days) for people to complain about this patch or more generally give feedback, and then let's merge. Feel free to ping maintainers of servant-foreign powered libraries if you know of any, as it might make their lives a little easier to know about this change beforehand.

File: servant-foreign/servant-foreign.cabal [hunk 1]
```diff
@@ -34,6 +34,7 @@ library
                      , lens       == 4.*
                      , servant    == 0.11.*
                      , text       >= 1.2  && < 1.3
+                     , http-media
```
Comment (jkarni) [inline servant-foreign/servant-foreign.cabal]: Could you include upper and lower bounds? (It's a mistake that we haven't done this for `http-types`.)

**Comment by tmbull** [comment]
Sorry for the delay. I was pretty busy last week and didn't have a chance to update this.

@jkarni, I wasn't sure what to do about the version number in the CHANGELOG so I just incremented the current minor version by 1. If that's not the correct version number, let me know and I'll fix it.

**Comment by alpmestan** [comment]
@jkarni @phadej Any other feedback?

**Comment by phadej** [comment]
Let's have this as part of 0.12. As @jkarni said it's LGTM, sans changelog (which is now there). I'll merge this very soon.

**Comment by phadej** [comment]
Superseded by #848

**Pull request closed without merging**